src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -144,7 +160,89 @@ public void Write(JsonTextWriter writer)
             return (Template.FromJson<Template>(content), content.FromJson<JToken>());
         }
 
-        private void EmitParametersIfPresent(JsonTextWriter jsonWriter, ExpressionEmitter emitter)
+        private void ProcessRawSourceMap(JToken rawTemplate)
+        {
+            var formattedTemplateLines = rawTemplate
+                .ToString(Formatting.Indented)
+                .Split(Environment.NewLine, StringSplitOptions.None);
+
+            // get line starts of unformatted JSON by stripping formatting from each line of formatted JSON
+            var unformattedLineStarts = formattedTemplateLines
+                .Aggregate(
+                    new List<int>() { 0 }, // first line starts at position 0
+                    (lineStarts, line) =>
+                    {
+                        var unformattedLine = JsonWhitespaceStrippingRegex.Replace(line, "$1");
```

Comment:
Instead of doing string replacement, can we format the template again with zero indentation (by setting `JsonTextWriter.Indentation = 0`) to calculate `unformattedLineStarts`?

Reply:
Thought this might work differently from `ToString(Formatting.None)` but it looks like this approach also strips newlines so it won't be possible to calculate the line starts.

    var stringWriter = new StringWriter();
    var jsonWriter = new JsonTextWriter(stringWriter);
    jsonWriter.Indentation = 0;
    rawTemplate.WriteTo(jsonWriter);
    var templateNoIndent = stringWriter.ToString(); // stripped of newlines